File: 백준/Silver/1946. 신입 사원/신입 사원.py
T = int(input())

for _ in range(T):
    N = int(input())
    s = []
    for i in range(N):
        a, b = map(int, input().split())
        s.append((a, b))

    s.sort()

    cnt = 1
    highrank = s[0][1]
    for i in range(N):
        if highrank > s[i][1]:
            cnt += 1
            highrank = s[i][1]

    print(cnt)

File: 백준/Silver/1946. 신입 사원/신입 사원.cc
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main()
{
    int T;
    cin >> T;
    while (T--)
    {
        int N;
        cin >> N;

        vector<pair<int, int>> s(N);
        for (int i = 0; i < N; i++)
        {
            cin >> s[i].first >> s[i].second;
        }

        sort(s.begin(), s.end());

        int cnt = 1;
        int highrank = s[0].second; // 면접 등수
        for (int i = 1; i < N; i++)
        {
            if (highrank > s[i].second)
            {
                cnt++;
                highrank = s[i].second;
            }
        }

        cout << cnt << endl;
    }
}